Engine/source/app/net/tcpObject.cpp

```diff
@@ -400,6 +405,29 @@ void TCPObject::send(const U8 *buffer, U32 len)
    Net::sendtoSocket(mTag, buffer, S32(len));
 }
 
+bool TCPObject::sendFile(const char* fileName)
+{
+	//Open the file for reading
+	FileStream readFile;
+	if(!readFile.open(fileName, Torque::FS::File::Read))
+	{
+		return false;
+	}
+
+	//Read each byte into our buffer
+	Vector<U8> buffer(readFile.getStreamSize());
+	U8 byte;
+	while(readFile.read(&byte))
+	{
+		buffer.push_back(byte);
+	}
+
+	//Send the buffer
+	send(buffer.address(), buffer.size());
```

Comment:
Oh also yeah the bytesWritten part I actually got from you code. My comment on running into an issue when it was queued was on there and I just send the entire file as it worked for what we needed it for. So they are somewhat related. My comments on it still stand though that you run into issues with that implementation of a send queue.